custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,183 @@
+"""Support for Somfy Heating Temperature Interface."""
+import logging
+from typing import Optional
+
+from homeassistant.components.climate import SUPPORT_PRESET_MODE, ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+OVERKIZ_TO_PRESET_MODES = {
+    OverkizCommandParam.SECURED: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.FREE: PRESET_NONE,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+OVERKIZ_TO_HVAC_MODES = {
+    OverkizCommandParam.AUTO: HVAC_MODE_AUTO,
+    OverkizCommandParam.MANU: HVAC_MODE_HEAT_COOL,
+}
+
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_HVAC_ACTION = {
+    OverkizCommandParam.COOLING: CURRENT_HVAC_COOL,
+    OverkizCommandParam.HEATING: CURRENT_HVAC_HEAT,
+}
+
+MAP_PRESET_TEMPERATURES = {
+    PRESET_COMFORT: OverkizState.CORE_COMFORT_ROOM_TEMPERATURE,
+    PRESET_ECO: OverkizState.CORE_ECO_ROOM_TEMPERATURE,
+    PRESET_AWAY: OverkizState.CORE_SECURED_POSITION_TEMPERATURE,
+}
+
+MODE_COMMAND_MAPPING = {
+    OverkizCommandParam.COMFORT: OverkizCommand.SET_COMFORT_TEMPERATURE,
+    OverkizCommandParam.ECO: OverkizCommand.SET_ECO_TEMPERATURE,
+    OverkizCommandParam.SECURED: OverkizCommand.SET_SECURED_POSITION_TEMPERATURE,
+}
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface.
+
+    The thermostat has 3 ways of working:
+      - Auto: Switch to eco/comfort temperature on a schedule (day/hour of the day)
+      - Manual comfort: The thermostat use the temperature of the comfort setting (19°C degree by default)
+      - Manual eco: The thermostat use the temperature of the eco setting (17°C by default)
+      - Freeze protection: The thermostat use the temperature of the freeze protection (7°C by default)
+
+    There's also the possibility to change the working mode, this can be used to change from a heated
+    floor to a cooling floor in the summer.
+
+    documentation: https://damrexelprod.blob.core.windows.net/medias/2fb79f22-2075-4f5a-93d3-112b14a4b8c2
```

Comment:
Let's see if we can add a codespell ignore somewhere here. Couldn't find the required # format yet on mobile.

Reply:
Or i can just remove the link.

Reply:
Fine by me as well :)